cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/ReadSegmentRangeCommand.java

```diff
@@ -42,29 +45,43 @@ public ReadSegmentRangeCommand(CommandArgs args) {
     }
 
     @Override
-    public void execute() throws ExecutionException, InterruptedException, TimeoutException {
-        ensureArgCount(4);
+    public void execute() throws ExecutionException, InterruptedException, TimeoutException, IOException {
+        ensureArgCount(5);
 
         final String fullyQualifiedSegmentName = getArg(0);
         final int offset = getIntArg(1);
         final int length = getIntArg(2);
         final String segmentStoreHost = getArg(3);
+        final String fileName = getArg(4);
         @Cleanup
         CuratorFramework zkClient = createZKClient();
         @Cleanup
         SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
         CompletableFuture<WireCommands.SegmentRead> reply = segmentHelper.readSegment(fullyQualifiedSegmentName,
                 offset, length, new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), "");
         WireCommands.SegmentRead segmentRead = reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
```

Comment:
This is a good start, however it will not work for more than 16MB at a time. There is a limit of 16MB that we can send over the wire at any given time (Netty frame size), so we'll need to make this handle larger sizes as well.

Consider doing this:
1. Open/Create the file
2. Enter a loop; at each iteration request a maximum of 8MB (half of 16MB). Each buffer you receive back will be written at the end of that file.
3. Loop until you have read all the range that you were asked for in the arguments.

This approach should also enable us to download more than 2GB at once (MaxInt). 

Tip: For a good user experience, consider outputting some information to the console every time you downloaded 8MB (every iteration). For example (I take this from `jhat`), output a dot (at once; do not `println`) for every 8MB. 

Example:
```
Downloading 16000001 bytes from offset 0. Expect 3 dots:
...
Wrote 16000001 bytes to /tmp/myfile.
```

In the above example, you have 3 dots because you break 16000001 into 3 parts: [0, 8000000), [8000000, 16000000) and [16000000, 16000001) (8MB+8MB+1B).

Reply:
`readSegment` has a `MAX_READ_SIZE` of 2MB. As we see here:
https://github.com/pravega/pravega/blob/f969bc907ad59bcef3387ae27021a08c57833502/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java#L210e
So we could reduce the buffer size.

Reply:
Also to add, `readSegment` call takes the offset and length as integers. While it makes sense to take them as `long` since as described the read will be as follows (offset, length).
```
(offset, 2000000) -> (offset + 2000000, 2000000) -> (offset + 400000, 2000000) -> ........
```
The length is fixed and shouldn't be an issue but since the offset can keep growing, should we change the offset argument in `ReadSegment` to a `long` type?

Reply:
I've implemented this approach and for the user info to be outputted I've added a byte counter of sorts which shows the amount data that has been processed and the total that's requested in this format:
```
Processing \ : Written 8000/160001 bytes.
```